### PR TITLE
feature/switch-to-JSONAPI-for-series-pages

### PR DESCRIPTION
**Pull request by jonhindle**
### Context

> Does this issue have a Trello card?

https://trello.com/c/EyX2RFzY/112-switch-to-jsonapi-for-series-pages

> If this is an issue, do we have steps to reproduce?

N/A

### Intent

> What changes are introduced by this PR that correspond to the above card?

We no longer serve series 

> Would this PR benefit from screenshots?

no

### Considerations

> Is there any additional information that would help when reviewing this PR?

original code left in for categories

> Are there any steps required when merging/deploying this PR?

no

### Checklist

- [ ] This PR contains **only** changes related to the above card
- [ ] Tests have been added/updated to cover the change
- [ ] Documentation has been updated where appropriate
- [ ] Tested in Development

### CODE REVIEW

File: server/repositories/cmsQueries/seriesHeaderPageQuery.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  transform(item) {
+    return {
+      id: item?.drupalInternal_Tid,
+      contentType: 'series',
+      name: item?.name,
+      description: item?.description?.processed,
+      image: {
+        url: item?.fieldFeaturedImage?.imageStyleUri[0]?.tile_large,
```
Comment (andrewrlee) [inline server/repositories/cmsQueries/seriesHeaderPageQuery.js:29]: I'm not sure the order of the images is safe to assume here.

Comment (andrewrlee) [reply server/repositories/cmsQueries/seriesHeaderPageQuery.js:29]: I wonder if Leon could modify the structure so it's easier to consume

File: server/repositories/cmsQueries/seriesPageQuery.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      contentUrl: `/content/${id}`,
+      image: {
+        url: item?.fieldMojThumbnailImage?.imageStyleUri[1].tile_small,
+        alt: item?.fieldMojThumbnailImage?.resourceIdObjMeta?.alt,
+      },
+    };
+  };
+
+  transform(deserializedResponse) {
+    return Object.assign(
```
Comment (andrewrlee) [inline server/repositories/cmsQueries/seriesPageQuery.js:69]: Meh.. I was doing a bit of reading around this.. It doesn't look like there is a good reason to use `Object.assign` for places where object spread would suffice. 

As well as spread being more succinct, I think it's generally good to prefer approaches that don't mutate existing data.

The only reason to use Object.assign seems to be if you want to actually call setters or preserve the prototype of the original object - would possibly only reserve it to be used in those cases.

File: server/routes/tags.js [hunk 3]
```diff
@@ -29,7 +29,9 @@ const createTagRouter = ({ hubTagsService }) => {
 
       data.secondaryTags = data.id;
 
-      const pageType = data.contentType === 'tags' ? 'tags' : 'tagsSeries';
+      const pageType = ['tags', 'series'].includes(data.contentType)
+        ? 'tags'
+        : 'tagsCategories';
```
Comment (andrewrlee) [inline server/routes/tags.js:34]: I thought we only supported secondary tags and series on this route? Do we need `tagsCategories`?

File: server/services/cms.js [hunk 4]
```diff
@@ -32,14 +38,25 @@ class CmsService {
...
+
+  async getTag(establishmentName, id) {
+    const lookupData = await this.#cmsApi.lookupTag(establishmentName, id);
     const { type, uuid, location } = lookupData;
     switch (type) {
       case 'taxonomy_term--tags':
         return this.getSecondaryTag(establishmentName, uuid, location);
       case 'taxonomy_term--series':
-        return null; // TODO
+        return this.getSeries(establishmentName, id, location);
```
Comment (andrewrlee) [inline server/services/cms.js:59]: we could possibly throw from default now if we come across some type of content we wouldn't expect to serve?

File: server/repositories/cmsQueries/homePageQuery.js [hunk 5]
```diff
@@ -49,10 +50,10 @@ class HomepageQuery {
 
   transformEach(item) {
     const [upperFeatured, lowerFeatured] = item.fieldMojFeaturedTileLarge.map(
-      this.#asTile('tile_large'),
+      this.#asTile(getLargeTile),
```
Comment (andrewrlee) [inline server/repositories/cmsQueries/homePageQuery.js:53]: meh.. maybe not worth holding this up but could possible get rid of the `uri` from the file fields as presumably no longer used?

File: server/repositories/cmsQueries/videoPageQuery.js [hunk 6]
```diff
@@ -57,11 +58,6 @@ class VideoPageQuery {
     }));
 
   transform(item) {
-    const url =
-      item.fieldMojThumbnailImage?.imageStyleUri
-        ?.map(image => image?.tile_large)
-        .find(image => Boolean(image)) || item.fieldMojThumbnailImage?.uri?.url;
```
Comment (andrewrlee) [inline server/repositories/cmsQueries/videoPageQuery.js:63]: same here regarding removing URI from fields.

File: server/repositories/cmsQueries/audioPageQuery.js [hunk 7]
```diff
@@ -58,11 +59,6 @@ class AudioPageQuery {
     }));
 
   transform(item) {
-    const url =
```
Comment (andrewrlee) [inline server/repositories/cmsQueries/audioPageQuery.js:61]: same here round uri

File: server/repositories/cmsQueries/seriesPageQuery.js (hunk 2 above)
Comment (jonhindle) [reply server/repositories/cmsQueries/seriesPageQuery.js]: The use of spread or assign here is negligible. The first object gets created and torn down and then recreated in this scenario with a spread, theoretically it only gets created once with Object.assign. I'm easy either way though 🤷

File: server/routes/tags.js (hunk 3 above)
Comment (jonhindle) [reply server/routes/tags.js:34]: In the interests of maintaining existing functionality, we should leave it. Raise a ticket to remove, that way any changes in functionality is coming from product.

File: server/services/cms.js (hunk 4 above)
Comment (jonhindle) [reply server/services/cms.js]: Agreed, but we still have to cater for categories atm.

File: server/repositories/cmsQueries/__tests__/homePageQuery.spec.js [hunk 8]
```diff
@@ -22,7 +22,9 @@ describe('Homepage query', () => {
             title: 'Lower Abs workout',
             fieldMojDescription: { summary: 'Intense lower core workout' },
             fieldMojThumbnailImage: {
-              uri: { url: 'https://cloud-platform-c3b3.eu-west-2' },
+              imageStyleUri: [
+                { tile_small: 'https://cloud-platform-c3b3.eu-west-2' },
+              ],
```
Comment (andrewrlee) [inline server/repositories/cmsQueries/__tests__/homePageQuery.spec.js:27]: couple of tests we could remove heree:
```
it('should show small tile image when provided', async () => {
it('should show large tile image when provided', async () => {
```

now we know that they will always exist

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**